src/basic_hierarchy/interfaces/Node.java

```diff
@@ -2,36 +2,94 @@
 
 import java.util.LinkedList;
 
-public interface Node {	
+
+/**
+ * A group is a collection of {@link Instance}s which belong to the same class.
```

Comment:
A {@link Node} is a collection of {@link Instance}s which were assign to one {@link Node} by i.e. some method or generator.